adapter/utils/src/main/java/org/mobilitydata/gtfsvalidator/timeutils/TimeUtilsImpl.java

```diff
@@ -118,4 +119,10 @@ public String convertIntegerToHMMSS(final Integer elapsedDurationSinceNoonInSeco
                     secondsAfterMidnight);
         }
     }
+
+    @Override
+    public boolean arePeriodsOverlapping(final int firstPeriodFirstTime, final int firstPeriodLastTime,
+                                         final int secondPeriodFirstTime, final int secondPeriodLastTime) {
+        return !(firstPeriodFirstTime <= secondPeriodLastTime && firstPeriodLastTime <= secondPeriodFirstTime);
```

Comment:
I believe this requires the same change as the frequency overlap logic to account for when the second period times occur before the first.

Reply:
Indeed